otter/util/retry.py

```diff
@@ -340,12 +340,14 @@ class Retry(object):
     :param should_retry: The function to call to determine whether retry
     should occur (usually an instance of :obj:`ShouldDelayAndRetry`).
     """
-    def perform_effect(self, dispatcher):
-        """
-        Invoke :func:`effect.retry.retry` with the effect and the
-        should_retry function.
-        """
-        return effect_retry(self.effect, self.should_retry)
+
+
+def perform_retry(dispatcher, intent):
```

Comment:
`dispatcher` is not used since this is not really _performing_ the effect. Maybe change the function to `invoke_effect_retry`?